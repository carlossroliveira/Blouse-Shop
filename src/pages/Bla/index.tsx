import React, { useState } from 'react';
import { Link } from 'react-router-dom';

/* import axios from 'axios'; */

export const Bla = () => {
  /* const [users, setUsers] = useState([]);

  useEffect(() => {
    axios
      .get('https://api.github.com/users/carlossroliveira/repos')
      .then((response) => response.data)
      .then((data) => setUsers(data));
  }, []); */

  const [posts, setPosts] = useState([]);

  React.useEffect(() => {
    fetch('http://localhost:3000/api/posts')
      .then((response) => response.json())
      .then((json) => setPosts(json));
  }, []);

  return (
    <>
      <ul>
        {posts.map((post: any) => (
          <>
            <li key={post.id}>
              <Link to={`/posts/${post.id}`}>{post.marca}</Link>
            </li>
          </>
        ))}
      </ul>
    </>
  );
};
