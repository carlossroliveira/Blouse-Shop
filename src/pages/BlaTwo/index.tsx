import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useParams } from 'react-router-dom';

export const BlaTwo = () => {
  const { id } = useParams();

  const { data } = useFetch<any>(`http://localhost:5000/posts/${id}`);

  return (
    <>
      <h2 className="cartao__titulo">{data?.marca}</h2>
      <h2 className="cartao__titulo">{data?.descricao}</h2>
    </>
  );
};
