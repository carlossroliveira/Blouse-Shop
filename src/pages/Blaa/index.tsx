import React from 'react';

export const Bla = () => {
  return (
    <>
      <h1>Bla bla</h1>
    </>
  );
};
