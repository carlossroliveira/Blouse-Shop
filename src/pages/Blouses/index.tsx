// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import { Link } from 'react-router-dom';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';

export const Blouses = () => {
  const { data } = useFetch<any>('http://localhost:5000/posts');

  return (
    <>
      <ul>
        {data?.map((post: any) => (
          <>
            <li key={post.id}>
              <Link to={`/posts/${post.id}`}>{post.marca}</Link>
            </li>
          </>
        ))}
      </ul>
    </>
  );
};
