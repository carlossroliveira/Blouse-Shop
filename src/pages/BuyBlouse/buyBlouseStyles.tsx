// -------------------------------------------------
// Packages
// -------------------------------------------------
import styled from 'styled-components';

export const ContainerSC = styled.section`
  display: flex;
  flex-wrap: wrap;
`;

export const DivSC = styled.div`
  gap: 1rem;
  display: flex;
  width: 60rem;
  margin: 0 auto;
`;
