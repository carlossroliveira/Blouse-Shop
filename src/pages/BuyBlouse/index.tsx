// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useEffect, useState } from 'react';
import { FaShippingFast } from 'react-icons/fa';
import { useNavigate, useParams } from 'react-router-dom';
import { BsArrowLeftCircle, BsArrowRightCircle } from 'react-icons/bs';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { NamePage } from '../../components/NamePage';
import { ButtonLink } from '../../components/ButtonLink';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { IBProps } from '../../utils/types';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import {
  ContainerSC,
  DivBuySC,
  DivFourSC,
  DivOneSC,
  DivSC,
  DivShieldSC,
  DivSizeSC,
  DivThreeSC,
  DivTwoSC,
  ImgSC,
  ImgShieldSC,
  ParagraphSC,
  SizeButtonSC,
} from './buyBlouseStyles';

export const BuyBlouse = () => {
  const { id } = useParams();
  const navigate = useNavigate();

  const [size, setSize] = useState<string | undefined>('');
  const [count, setCount] = useState<number>(Number(id) + 1);

  const { data } = useFetch<IBProps>(`http://localhost:5000/posts/${id}`);

  const verifyIsSelected = (selected: string) => size === selected;

  const leftFunction = () => count >= 1 && setCount(count - 1);

  const rightFunction = () => count <= 9 && setCount(count + 1);

  useEffect(() => {
    setSize('');
    navigate(`/posts/0${count}`);
  }, [count, navigate]);

  return (
    <ContainerSC>
      <NamePage title="Camisas" />

      <DivSC>
        <DivOneSC>
          <button
            disabled={Number(id) === 1 ? true : false}
            onClick={leftFunction}
          >
            <BsArrowLeftCircle />
          </button>
          <ImgSC
            src={`${data?.foto?.map((item) => item.src)}`}
            alt="Camisa de time"
          />
          <button
            disabled={Number(id) === 9 ? true : false}
            onClick={rightFunction}
          >
            <BsArrowRightCircle />
          </button>
        </DivOneSC>

        <DivTwoSC>
          <DivFourSC>
            {data?.info && (
              <ParagraphSC Info>
                <FaShippingFast /> {data?.info}
              </ParagraphSC>
            )}

            <ParagraphSC Title>
              {data?.foto?.map((item) => item.titulo)}
            </ParagraphSC>

            {data?.preco?.map((item, index) => (
              <>
                <ParagraphSC Inactive key={index}>
                  R$ {item.valorInativo}
                </ParagraphSC>

                <ParagraphSC Installments>R$ {item.valorInicial}</ParagraphSC>

                <ParagraphSC InstallmentsTwo>
                  Ou {item.valorIniParcelado}x de R$ {item.valorInaParcelado}{' '}
                  sem juros
                </ParagraphSC>
              </>
            ))}
          </DivFourSC>

          <DivShieldSC>
            {data?.foto?.map((item) => (
              <ImgShieldSC
                key={item.titulo}
                src={item.shield}
                alt="Escudo do time"
              />
            ))}
          </DivShieldSC>

          <DivThreeSC>
            <ParagraphSC>
              <span>Tamanho: {size}</span>

              <DivSizeSC>
                {data?.tamanho?.slice(0, 1) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(0, 1))}
                    onClick={() => setSize(data?.tamanho?.slice(0, 1))}
                  >
                    {data?.tamanho?.slice(0, 1)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(3, 4) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(3, 4))}
                    onClick={() => setSize(data?.tamanho?.slice(3, 4))}
                  >
                    {data?.tamanho?.slice(3, 4)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(6, 7) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(6, 7))}
                    onClick={() => setSize(data?.tamanho?.slice(6, 7))}
                  >
                    {data?.tamanho?.slice(6, 7)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(9, 11) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(9, 11))}
                    onClick={() => setSize(data?.tamanho?.slice(9, 11))}
                  >
                    {data?.tamanho?.slice(9, 11)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(13, 16) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(13, 16))}
                    onClick={() => setSize(data?.tamanho?.slice(13, 16))}
                  >
                    {data?.tamanho?.slice(13, 16)}
                  </SizeButtonSC>
                )}

                {data?.tamanho?.slice(18, 22) && (
                  <SizeButtonSC
                    Background={verifyIsSelected(data?.tamanho?.slice(18, 22))}
                    onClick={() => setSize(data?.tamanho?.slice(18, 22))}
                  >
                    {data?.tamanho?.slice(18, 22)}
                  </SizeButtonSC>
                )}
              </DivSizeSC>
            </ParagraphSC>

            <ParagraphSC>
              <span>Tercido:</span> {data?.tecido}
            </ParagraphSC>

            <ParagraphSC>
              <span>Marca:</span> {data?.marca}
            </ParagraphSC>

            <DivBuySC>
              <ParagraphSC>
                <span>Descrição:</span> {data?.descricao}
              </ParagraphSC>

              <ButtonLink
                text="Comprar"
                onClick={() => alert('Comprar')}
                color="black"
              />

              <button onClick={() => alert('Comprar')}>aaaaaaaa</button>
            </DivBuySC>
          </DivThreeSC>
        </DivTwoSC>
      </DivSC>
    </ContainerSC>
  );
};

/* 

position: absolute;
    width: 30rem;
    right: -1rem;
    top: 7rem; 
    
    */
