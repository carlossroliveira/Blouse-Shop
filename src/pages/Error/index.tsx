// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { ButtonLink } from '../../components/ButtonLink';
import { SocialNetworks } from '../../components/SocialNetworks';
// -------------------------------------------------
// Images
// -------------------------------------------------
import photo from './OopsError.svg';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import {
  ContainerSC,
  DivOneSC,
  DivSC,
  DivTwoSC,
  ParagraphSC,
  TitleSC,
} from './errorStyles';

export const Error = (): JSX.Element => {
  return (
    <ContainerSC>
      <DivOneSC>
        <img src={photo} alt="image error" />
      </DivOneSC>

      <DivTwoSC>
        <TitleSC>Esta página não existe!</TitleSC>

        <ParagraphSC>Clique no botão para volta</ParagraphSC>

        <ButtonLink text="Voltar" path="/" />

        <DivSC>
          <SocialNetworks />
        </DivSC>
      </DivTwoSC>
    </ContainerSC>
  );
};
