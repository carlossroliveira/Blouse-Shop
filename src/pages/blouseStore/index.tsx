import React, { useEffect } from 'react';

import axios from 'axios';
import { Mirage } from '../../mirage';

export const BlouseStore = () => {
  /* const [users, setUsers] = useState([]); */

  useEffect(() => {
    axios.get('api/users').then((response) => response.data);
    /* .then((data) => setUsers(data)); */
  }, []);

  Mirage();

  return (
    <>
      <h1>Testing Mirage JS</h1>

      <ul>
        {/* {users?.map((user: any) => (
          <>
            <img
              src={user?.foto?.map((fotos: any) => fotos.src)}
              alt="imagem"
            />
            <div key={user.id}>
              <p>{user?.foto?.map((fotos: any) => fotos.titulo)}</p>

              <p>{user.tecido}</p>
              <p>{user.tamanho}</p>
              <p>{user.preco}</p>
              <p>{user.marca}</p>
              <p>{user.descricao}</p>
            </div>
          </>
        ))} */}
      </ul>
    </>
  );
};
