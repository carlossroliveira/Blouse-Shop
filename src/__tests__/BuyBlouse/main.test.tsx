// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { BuyBlouse } from '../../pages/BuyBlouse';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../../utils/withThemeComponent';

const LoginWithTheme = WithThemeComponent(BuyBlouse);

describe('Testing error buyBlouse', () => {
  it('Should render some information correctly', () => {
    render(<LoginWithTheme />);

    expect(
      screen.getByRole('img', {
        name: /camisa de time/i,
      }),
    ).toBeInTheDocument();

    expect(screen.getByText(/tamanho:/i)).toBeInTheDocument();

    expect(screen.getByText(/tercido:/i)).toBeInTheDocument();

    expect(screen.getByText(/marca:/i)).toBeInTheDocument();

    expect(screen.getByText(/descrição:/i)).toBeInTheDocument();

    expect(
      screen.getByRole('button', {
        name: /comprar agora/i,
      }),
    ).toBeInTheDocument();
  });

  it('component rendering matches snapshot correctly', () => {
    render(<LoginWithTheme />);
    expect(render(<LoginWithTheme />)).toMatchSnapshot();
  });
});
