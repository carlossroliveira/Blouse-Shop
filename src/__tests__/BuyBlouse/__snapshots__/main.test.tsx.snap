// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing error buyBlouse component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c13 {
  width: 40rem;
  height: 4rem;
  background-color: #8ca315;
  border: none;
  font-weight: 700;
  border-radius: 5px;
  text-transform: uppercase;
  box-shadow: rgb(38,57,77) 0px 20px 30px -10px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c13:hover {
  cursor: pointer;
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  gap: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0rem 3rem;
}

.c4 {
  width: 30rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border: none;
  font-size: 2rem;
  background-color: transparent;
}

.c3:hover {
  cursor: pointer;
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c3 svg {
  color: #8ca315;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 500px;
  -ms-flex: 1 500px;
  flex: 1 500px;
  position: relative;
}

.c7 {
  color: black;
  color: #383838;
  font-size: 1.2rem;
  font-weight: 700;
}

.c7 span {
  color: #383838;
  font-weight: bold;
}

.c10 {
  color: black;
}

.c10 span {
  color: #383838;
  font-weight: bold;
}

.c6 {
  margin-top: 0px;
}

.c9 {
  margin-top: 0px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 5px;
}

.c8 {
  position: absolute;
  right: 1rem;
  top: -1rem;
}

.c12 {
  gap: 5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

.c12 section {
  margin-right: 6rem;
}

@media (max-width:975px) {
  .c13 {
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
  }
}

@media (max-width:975px) {
  .c4 {
    width: 100%;
  }
}

@media (max-width:1520px) {
  .c2 {
    margin: 0 auto;
  }
}

@media (max-width:975px) {
  .c8 {
    display: none;
  }
}

@media (max-width:975px) {
  .c12 {
    gap: 1rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<body>
    <div>
      <section
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <button
              class="c3"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <img
              alt="Camisa de time"
              class="c4"
              src="undefined"
            />
            <button
              class="c3"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <p
                class="c7"
              />
            </div>
            <div
              class="c8"
            />
            <div
              class="c9"
            >
              <p
                class="c10"
              >
                <span>
                  Tamanho: 
                </span>
                <div
                  class="c11"
                />
              </p>
              <p
                class="c10"
              >
                <span>
                  Tercido:
                </span>
                 
              </p>
              <p
                class="c10"
              >
                <span>
                  Marca:
                </span>
                 
              </p>
              <div
                class="c12"
              >
                <p
                  class="c10"
                >
                  <span>
                    Descrição:
                  </span>
                   
                </p>
                <button
                  class="c13"
                >
                  Comprar agora
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="ReactModalPortal"
    />
    <div>
      <section
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <button
              class="c3"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <img
              alt="Camisa de time"
              class="c4"
              src="undefined"
            />
            <button
              class="c3"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <p
                class="c7"
              />
            </div>
            <div
              class="c8"
            />
            <div
              class="c9"
            >
              <p
                class="c10"
              >
                <span>
                  Tamanho: 
                </span>
                <div
                  class="c11"
                />
              </p>
              <p
                class="c10"
              >
                <span>
                  Tercido:
                </span>
                 
              </p>
              <p
                class="c10"
              >
                <span>
                  Marca:
                </span>
                 
              </p>
              <div
                class="c12"
              >
                <p
                  class="c10"
                >
                  <span>
                    Descrição:
                  </span>
                   
                </p>
                <button
                  class="c13"
                >
                  Comprar agora
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": .c13 {
  width: 40rem;
  height: 4rem;
  background-color: #8ca315;
  border: none;
  font-weight: 700;
  border-radius: 5px;
  text-transform: uppercase;
  box-shadow: rgb(38,57,77) 0px 20px 30px -10px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c13:hover {
  cursor: pointer;
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  gap: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0rem 3rem;
}

.c4 {
  width: 30rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border: none;
  font-size: 2rem;
  background-color: transparent;
}

.c3:hover {
  cursor: pointer;
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c3 svg {
  color: #8ca315;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 500px;
  -ms-flex: 1 500px;
  flex: 1 500px;
  position: relative;
}

.c7 {
  color: black;
  color: #383838;
  font-size: 1.2rem;
  font-weight: 700;
}

.c7 span {
  color: #383838;
  font-weight: bold;
}

.c10 {
  color: black;
}

.c10 span {
  color: #383838;
  font-weight: bold;
}

.c6 {
  margin-top: 0px;
}

.c9 {
  margin-top: 0px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 5px;
}

.c8 {
  position: absolute;
  right: 1rem;
  top: -1rem;
}

.c12 {
  gap: 5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

.c12 section {
  margin-right: 6rem;
}

@media (max-width:975px) {
  .c13 {
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
  }
}

@media (max-width:975px) {
  .c4 {
    width: 100%;
  }
}

@media (max-width:1520px) {
  .c2 {
    margin: 0 auto;
  }
}

@media (max-width:975px) {
  .c8 {
    display: none;
  }
}

@media (max-width:975px) {
  .c12 {
    gap: 1rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
    <section
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <button
            class="c3"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <img
            alt="Camisa de time"
            class="c4"
            src="undefined"
          />
          <button
            class="c3"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <p
              class="c7"
            />
          </div>
          <div
            class="c8"
          />
          <div
            class="c9"
          >
            <p
              class="c10"
            >
              <span>
                Tamanho: 
              </span>
              <div
                class="c11"
              />
            </p>
            <p
              class="c10"
            >
              <span>
                Tercido:
              </span>
               
            </p>
            <p
              class="c10"
            >
              <span>
                Marca:
              </span>
               
            </p>
            <div
              class="c12"
            >
              <p
                class="c10"
              >
                <span>
                  Descrição:
                </span>
                 
              </p>
              <button
                class="c13"
              >
                Comprar agora
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
